### ISSUE TITLE
🚀 Feature: Get random document from a given collection

### ISSUE THREAD

**Issue by byawitz** [appwrite/appwrite#5205]
### 🔖 Feature description

It would be great if the database would have either function to get random as follow 
```dart
await database.getRandom([database_id], [collection_id])
```

Or by adding it in the query option, something like this:
```dart
databases.listDocuments([database_id], [collection_id], queries: [
        Query.by(Query.random()),
        Query.limit(4),
      ]);
```

The second approach will give the advantage to get a few random documents.

BTW: The best solution to achieve such info is to know the Collection length then just set random offset based on it
```dart
databases.listDocuments([database_id], [collection_id], queries: [
        Query.offset(random(fromTableLength())),
        Query.limit(4),
      ]);
```
And by @stnguyen90 idea I've also added a triggered function that will update the collection length in another document.

### 🎤 Pitch

In-game development case, when the level challenge is randomize by the AppWrite endpoint

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by drmittelstein** [comment]
I strongly support this idea!  Very important functionality that would greatly benefit AppWrite

**Comment by riteshreg** [comment]
This query will help a lot

**Comment by polbuhanky** [comment]
Is there any progress on this topic? Very necessary functionality

**Comment by ItzNotABug** [comment]
Subscribing to this!

**Comment by sylvainjack** [comment]
Yes I clearly would love such a feature too. I need it badly for my app.

**Comment by katriellucas** [comment]
I want to add another use that even the current work around is not suitable, in cases where you have to filter a random doc, for example, I have "stages" data that are labeled by difficulty, and I want only to get _HARD_ stages, that is not possible with the offset approach. I would have to make counters for each paramameter that I add, it's just not viable.

**Comment by AKRking** [comment]
> Yes I clearly would love such a feature too. I need it badly for my app. But what would be really great is to be able to combine RANDOM and some filters : For example : Give me x documents randomly corresponding to this condition (equal to, not equal to....). 

What did u do to achieve it? My soul of app depends on random doc since its a mock test app

**Comment by katriellucas** [comment]
> > Yes I clearly would love such a feature too. I need it badly for my app. But what would be really great is to be able to combine RANDOM and some filters : For example : Give me x documents randomly corresponding to this condition (equal to, not equal to....).
> 
> What did u do to achieve it? My soul of app depends on random doc since its a mock test app

Well, i coudn't achieve it at all, I thought of some ideas that could work, but in the end I decided to not use Appwrite, instead I started using https://github.com/pocketbase/pocketbase

**Comment by riteshreg** [comment]
wow it's almost 3 years since I commented.

My suggestion is to use a sequential ID as the primary key. At runtime, retrieve the total count of documents in the table, then generate as many random numbers as needed—ensuring none exceed the total count. Use those random numbers as IDs to fetch the corresponding records.

**Comment by abnegate** [comment]
This feature is now available on Appwrite Cloud and in latest SDK's as `Query.orderRandom()`
